PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/ElementalTypes/ElementalTypeSheetRowParserTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿using FluentAssertions;
+using PokeOneWeb.DataSync.GoogleSpreadsheet.Import.Impl;
+using PokeOneWeb.DataSync.GoogleSpreadsheet.Import.Impl.Sheets.ElementalTypes;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace PokeOneWeb.DataSync.Tests.GoogleSpreadsheet.Import.Impl.Sheets.ElementalTypes
+{
+    public class ElementalTypeSheetRowParserTest
+    {
+        [Fact]
+        public void ReadRow_WithMinimalValidValues_ShouldParse()
+        {
+            // Arrange
+            var parser = new ElementalTypeSheetRowParser();
+            var name = "Name";
+            var values = new List<object> { name };
+
+            var expected = new ElementalTypeSheetDto()
+            {
+                Name = name
+            };
+
+            // Act
+            var result = parser.ReadRow(values);
+
+            // Assert
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void ReadRow_WithInsufficientValues_ShouldThrow()
+        {
+            // Arrange
+            var parser = new ElementalTypeSheetRowParser();
+            var values = new List<object>();
+
+            // Act
+            var actual = () => parser.ReadRow(values);
+
+            // Assert
+            actual.Should().Throw<InvalidRowDataException>();
+        }
+
+        [Fact]
+        public void ReadRow_WithTooManyValues_ShouldThrow()
+        {
+            // Arrange
+            var parser = new ElementalTypeSheetRowParser();
+            var values = new List<object>
+            {
+                "0", "excessive value"
+            };
+
+            // Act
+            var actual = () => parser.ReadRow(values);
+
+            // Assert
+            actual.Should().Throw<InvalidRowDataException>();
+        }
+
+        [Fact]
+        public void ReadRow_WithValuesNull_ShouldThrow()
+        {
+            // Arrange
+            var parser = new ElementalTypeSheetRowParser();
+            List<object> values = null;
+
+            // Act
+            var actual = () => parser.ReadRow(values);
+
+            // Assert
+            actual.Should().Throw<InvalidRowDataException>();
+        }
+
+        [Theory]
+        [InlineData("")] // Name must be non-empty
```

Comment:
Could also be a Fact if only one inline data is to be tested. This is also in other test files like this.

Reply:
True, will adjust that.